### ISSUE TITLE
Translate the gradle plugin documentation samples to Kotlin

### ISSUE THREAD

**Issue by jnizet** [spring-projects/spring-boot#14559]
The Gradle Kotlin DSL is almost at version 1.0 now, and all the samples of the Gradle user guide have been translated to Kotlin (see the nightly release of the user guide, for example https://docs.gradle.org/nightly/userguide/building_java_projects.html)

The Gradle team [would like popular plugins to show examples in Kotlin](https://github.com/gradle/gradle/issues/6790), too.

I have translated a lot of the samples of the Gradle user guide, and would like to help translating the samples of the Spring Boot Gradle plugin, too.

My plan would be to use tabs, just like the gradle user guide does, to show Groovy and Kotlin samples side by side in the generated documentation. I would use the [Spring Asciidoctor extensions](https://github.com/spring-io/spring-asciidoctor-extensions) for these tabs.

Would that be OK for you? Is there something I should know before starting working on this task?

**Comment by wilkinsona** [comment]
> Would that be OK for you?

It certainly would. Thank you for the offer.

> Is there something I should know before starting working on this task?

One thing that might not be immediately obvious is that we have tests for the build scripts that get included in the documentation. You can find the scripts [here](https://github.com/spring-projects/spring-boot/tree/f1a38c461f56a48740f40cd9a28a3f99c35316a4/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/gradle) and the tests [here](https://github.com/spring-projects/spring-boot/tree/f1a38c461f56a48740f40cd9a28a3f99c35316a4/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/test/java/org/springframework/boot/gradle/docs). If possible, I'd like to follow this pattern with the Kotlin DSL examples.

**Comment by wilkinsona** [comment]
I've assigned this to 2.x for now. If a pull request is made before 2.1 is released, then I'll gladly reassign it to an earlier milestone.

**Comment by wilkinsona** [comment]
Closing in favour of #14585.
